subiquity/models/subiquity.py

```diff
@@ -278,6 +280,12 @@ def _cloud_init_config(self):
             if getattr(model, 'make_cloudconfig', None):
                 merge_config(config, model.make_cloudconfig())
         merge_cloud_init_config(config, self.userdata)
+        # Raises cloudinit.config.schema.SchemaValidationError or
+        # SchemaDeprecationError to alert to merged-cloud-config issues
+        # in installer environment instead of after installed target reboot.
+        validate_cloudconfig_schema(
+            config, strict=True, log_details=False, log_deprecations=True
+        )
```

Comment:
One issue to note is that we would be running this validation agains the current cloud-init installed in the ephemeral environment, not the cloud-init version installed in the target environment. I've seen minor revision drift a few times between the ephemeral environment version of cloud-init and the target  environment that is being installed. While it is unlikely that there is enough of a drift to deprecate or drop keys between those minor versions, there could be bug fixes related to schema that could crop up resulting in invalid schema warnings/errors. Is it worth us ensuring that we source and run the config validation within the target environment (maybe via chroot). Or cloud-init to provide a mechanism to ensure this entry point is confined to only schema files within the `/target` path

Reply:
Something that may help the target environment is that updates should be applied before target environment first boot.  No guarantees on that front, though.

It's pretty corner case, but if someone used a original version LTS iso with no point release updates, allowed Subiquity to update, then blocked other updates from running, that could be a pretty large (years) version drift.   But, someone blocking updates probably isn't allowing Subiquity to update either.

I suggest we start without chroot into target for schema validation artifacts and see how it goes.